### PR TITLE
[ShapeOpt] Corrected gid design logger

### PR DESCRIPTION
**Pull request by dbaumgaertner**
introduced forgotten recursive validation and assigns nodal results in a correct way

### CODE REVIEW

File: applications/ShapeOptimizationApplication/python_scripts/design_logger_gid.py [hunk 1]
```diff
@@ -64,25 +63,12 @@ def __DetermineOutputMode( self ):
...
-        elif self.write_optimization_model_part:
-            self.gid_io = GiDOutputProcess(self.optimization_model_part, design_history_file_path, gid_config)
-
     # --------------------------------------------------------------------------
     def __ModifySettingsToMatchDefaultGiDOutputProcess( self ):
         gid_parameters = self.output_settings["output_format"]["gid_parameters"]
 
         # Add nodal results
-        nodal_results = self.output_settings["nodal_results"]
-        self.output_settings["output_format"]["gid_parameters"]["result_file_configuration"].AddValue("nodal_results", nodal_results)
```
Comment (jcotela) [inline applications/ShapeOptimizationApplication/python_scripts/design_logger_gid.py:85]: @RiccardoRossi @loumalouomega please check out this specific change. @dbaumgaertner was using `AddValue` with an entry that already exists. Rapidjson used to accept it and replace the values, while the new json library skips it (it finds that an entry with that name already exists, and just exits without changing anything). Is this a known Behavior Change?

Comment (loumalouomega) [reply applications/ShapeOptimizationApplication/python_scripts/design_logger_gid.py:85]: That part of the code in particular was implemented originally by @RiccardoRossi fortunately can be easily solved

Comment (loumalouomega) [reply applications/ShapeOptimizationApplication/python_scripts/design_logger_gid.py:85]: Done

Comment (jcotela) [reply applications/ShapeOptimizationApplication/python_scripts/design_logger_gid.py:85]: Not saying it is a bug, just that neither @dbaumgaertner nor me where aware of the change. In any case, thanks for the quick patch :+1:

**Review by armingeiser — APPROVED** [review]

**Pull request merged**